section.go

```diff
@@ -0,0 +1,60 @@
+//	Copyright (c) 2023 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zap
+
+import (
+	"github.com/RoaringBitmap/roaring"
+	index "github.com/blevesearch/bleve_index_api"
+)
+
+type section interface {
+	// process is essentially parsing of a specific field's content in a specific
+	// document. any tracking of processed data *specific to this section* should
+	// be done in opaque which will be passed to the Persist() API.
+	Process(opaque map[int]resetable, docNum uint64, f index.Field, fieldID uint16)
```

Comment:
docNum datatype should be uint32, because we use a uint32 roaring bitmap to track deleted documents, so we should keep the docNum datatype consistent.

Reply:
the main reason why i kept it as uint64 was because it was like that in the original code, however i think this makes sense to me.